### PR TITLE
feat(route): 添加 dm_img_list 字段防止被 Bilibili 识别

### PR DESCRIPTION
**Pull request by NightSpaceC**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/bilibili/user/video/12590
/bilibili/user/video-all/12590
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [ ] New Route / 新的路由
  - [ ] Follows [v2 Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [v2 路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Documentation / 文档说明
- [ ] Full text / 全文获取
  - [ ] Use cache / 使用缓存
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [ ] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [ ] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
我在请求参数中添加了非空的 `dm_img_list` 字段

`/bilibili/user/video-all` 仍然可用，我重新添加了它

代码中 `'[{"x":2721,"y":615,"z":0,"timestamp":29,"type":0}]'` 是我从浏览器中获取的一串可能的数据

也可以通过 `BILIBILI_DM_IMG_LIST` 环境变量自行添加

尽管它可以工作，我仍然不清楚 `dm_img_list` 的含义

它也许和鼠标移动有关，因为当我不移动鼠标时其值为 `[]` (空列表)

我无法随机生成它来欺骗系统，Bilibili 也许引入了机器学习来分辨人类与机器

### CODE REVIEW

File: lib/v2/bilibili/utils.js [hunk 1]
```diff
@@ -65,20 +66,35 @@ function hexsign(e) {
...
+function getDmImgList() {
+    if (typeof (config.bilibili.dmImgList) === 'undefined') {
+        return '[{"x":2721,"y":615,"z":0,"timestamp":29,"type":0}]';    // This is true data from browser
+    }
+    return config.bilibili.dmImgList;
+}
+
+function addDmVerifyInfo(params, dm_img_list) {
+    const dm_img_str = Buffer.from('no webgl').toString('base64').slice(0, -2);
+    const dm_cover_img_str = Buffer.from('no webgl').toString('base64').slice(0, -2);
```
Comment (TonyRL) [inline lib/v2/bilibili/utils.js:87]: `camelCase`

File: lib/v2/bilibili/utils.js [hunk 2]
```diff
@@ -65,20 +66,35 @@ function hexsign(e) {
...
     const verifyParam = searchParams.toString();
     const wts = Math.round(Date.now() / 1000);
-    const w_rid = md5(`${verifyParam}&wts=${wts}${verifyString}`);
+    const w_rid = md5(`${verifyParam}&wts=${wts}${wbiVerifyString}`);
     return `${params}&w_rid=${w_rid}&wts=${wts}`;
 }
 
+function getDmImgList() {
+    if (typeof (config.bilibili.dmImgList) === 'undefined') {
+        return '[{"x":2721,"y":615,"z":0,"timestamp":29,"type":0}]';    // This is true data from browser
```
Comment (TonyRL) [inline lib/v2/bilibili/utils.js:80]: They are simply the cursor locations of which these events happened.
<details><summary>Events</summary>

```js
var p = ["click", "mousemove", "keydown", "DOMMouseScroll", "mousewheel", "mousedown", "touchstart", "touchmove", "focus"]
```
</details>

<details><summary>log stack</summary>

```js
F = function(t) {
        var e = N
          , n = function(t, e, n, r, o, i) {
            return t(e, n, r, o, i)
        }
          , r = t[e(519)]((function(t, e) {
            var r = t.x
              , o = void 0 === r ? 0 : r
              , i = t.y
              , a = void 0 === i ? 0 : i
              , c = t.timestamp
              , u = t.type;
            return n(B, o, a, c, u, e)
        }
        ))[e(519)]((function(t) {
            return {
                x: t[0],
                y: t[1],
                z: t[2],
                timestamp: t[3],
                type: t[4]
            }
        }
        ));
        try {
            return JSON.stringify(r)
        } catch (t) {
            return console[e(541)](t),
            ""
        }
    }
```
</details>

I would prefer generating a random one instead.

File: lib/v2/bilibili/utils.js (hunk 1 above)
Comment (NightSpaceC) [reply lib/v2/bilibili/utils.js]: fixed

File: lib/v2/bilibili/utils.js (hunk 2 above)
Comment (NightSpaceC) [reply lib/v2/bilibili/utils.js:80]: Sorry, I really don't know how to generate it correctly.

I have tried to generate it from Gaussian Distributed, but it didn't work.

```js
// https://github.com/errcw/gaussian/blob/master/lib/box-muller.js
function generateGaussianInteger(mean, std) {
    const _2PI = Math.PI * 2;
    const u1 = Math.random();
    const u2 = Math.random();

    const z0 = Math.sqrt(-2.0 * Math.log(u1)) * Math.cos(_2PI * u2);

    return Math.round(z0 * std + mean);
}

function getDmImgList() {
    const path = [
        {
            x: Math.max(generateGaussianInteger(2800, 100), 0),
            y: Math.max(generateGaussianInteger(1000, 200), 0),
            z: 0,
            timestamp: Math.max(generateGaussianInteger(50, 5), 0),
            type: 0,
        },
    ];
    return JSON.stringify(path);
    // if (typeof (config.bilibili.dmImgList) === 'undefined') {
    //     return '[{"x":2721,"y":615,"z":0,"timestamp":29,"type":0}]';    // This is true data from browser
    // }
    // return config.bilibili.dmImgList;
}
```

Comment (TonyRL) [reply lib/v2/bilibili/utils.js:80]: I think the hints are quite obvious in https://github.com/DIYgod/RSSHub/pull/14128#discussion_r1437770620.
<details><summary>Anyways, these are the related codes dumped from 9.space.*.js</summary>

```js
function G() {
    var t = ["szSez", "103591TJTTxc", "CjXyY", "CRCsi", "handler", "length", "map", "TXMjE", "189vkvYeF", "AWumG", "HwlZR", "init", "324378lGstCz", "valueMap", "WKvHl", "name", "3716570VOnHWp", "TBybc", "wcSXW", "mhLTH", "webglStr", "14nRDWOh", "error", "AioYO", "buffer", "2550606OYIlVp", "keys", "XGGLO", "log", "params", "KfRTB", "AHjCN", "92152cwuYJw", "rAndRender", "bdUwC", "webglVendo", "18zWDGOL", "tkpid", "stringify", "encode", "LOG", "payload", "forEach", "apply", "eFIXr", "34444817xwEJrQ", "ytHiH", "request", "hQAjN", "LsOHu", "substring", "userLogStr", "CdcgQ", "PsWqo", "MmJfn", "OdIfM", "ImqLE", "DhwGv", "4274880qQISqI", "qmlRV", "LztiW", "jmfws", "queryUserL", "ZHgrM", "4KGZfal"];
    return (G = function() {
        return t
    }
    )()
}

var N = M;
function M(t, e) {
    var n = G();
    return (M = function (t, e) {
        return n[(t -= 498)];
    })(t, e);
}

var D = {
    mousemove: 0,
    click: 1,
    keydown: 2,
    wheel: 3,
    touch: 4,
    focus: 5,
};

function F(t) {
    var e = N,
        n = function (t, e, n, r, o, i) {
            return t(e, n, r, o, i);
        },
        r = t
            .map(function (t, e) {
                var r = t.x,
                    o = void 0 === r ? 0 : r,
                    i = t.y,
                    a = void 0 === i ? 0 : i,
                    c = t.timestamp,
                    u = t.type;
                return n(B, o, a, c, u, e);
            })
            .map(function (t) {
                return { x: t[0], y: t[1], z: t[2], timestamp: t[3], type: t[4] };
            });
    try {
        return JSON.stringify(r);
    } catch (t) {
        return console.log(t), '';
    }
}

function B(t, e, n, r, o) {
    var i = N,
        a = {
            CRCsi: function (t, e) {
                return t(e);
            },
            add: function (t, e) {
                return t + e;
            },
            add_1: function (t, e) {
                return t + e;
            },
            multiply: function (t, e) {
                return t * e;
            },
            minus: function (t, e) {
                return t - e;
            },
            CjXyY: function (t, e) {
                return t(e);
            },
            OdIfM: function (t, e) {
                return t(e);
            },
        },
        c = a.CRCsi(H, o);
    return [
        a.add(a.add_1(a.CRCsi(V, 1) * t, a.multiply(a.CRCsi(V, 0), e)), c),
        a.add(a.minus(a.CjXyY(V, 2) * t, a.multiply(a.OdIfM(V, 3), e)), c),
        c,
        n,
        D[r],
    ];
}

var W = ['Mg', 'Mw', 'NA', 'NQ']; // 2, 3, 4, 5
function V(t) {
    var e = N,
        n = {
            add: function (t, e) {
                return t + e;
            },
            bdUwC: function (t, e) {
                return t(e);
            },
        },
        r = n.add(W[t], '==');
    return n.bdUwC(Number, Buffer.from(r, 'base64').toString('ascii'));
}

function H(t) {
    var e = N,
        n = {
            multiply: function (t, e) {
                return t * e;
            },
        };
    return Math.floor(n.multiply(n.multiply(Math.random(), 114), t));
}

console.log(
    F([
        {
            type: 'mousemove',
            x: 0,
            y: 0,
            preX: 0,
            preY: 0,
            changeDistance: 0,
            timestamp: 0,
        },
    ]),
);
```
</details>

---

<details><summary>Simplified version</summary>

```js
const D = {
    mousemove: 0,
    click: 1,
    keydown: 2,
    wheel: 3,
    touch: 4,
    focus: 5,
};

const F = (t) => {
    var n = (func, x, y, timestamp, type, index) => {
        return func(x, y, timestamp, type, index);
    };
    var r = t
        .map((e, i) => {
            const x = e.x || 0;
            const y = e.y || 0;
            const timestamp = e.timestamp;
            const type = e.type;
            return n(B, x, y, timestamp, type, i);
        })
        .map((t) => {
            return { x: t[0], y: t[1], z: t[2], timestamp: t[3], type: t[4] };
        });
    try {
        return JSON.stringify(r);
    } catch (t) {
        return console.log(t), '';
    }
};

const B = (x, y, timestamp, type, index) => {
    const c = H(index);
    return [V[1] * x + V[0] * y + c, V[2] * x - V[3] * y + c, c, timestamp, D[type]];
};

// var W = ['Mg', 'Mw', 'NA', 'NQ'];
const V = [2, 3, 4, 5];

const H = (index) => {
    return Math.floor(Math.random() * 114 * index);
};

console.log(
    F([
        {
            type: 'mousemove',
            x: 0,
            y: 100,
            preX: 0,
            preY: 0,
            changeDistance: 0,
            timestamp: 0,
        },
    ]),
);
```
</details>

**Pull request merged**